client/client.go

```diff
@@ -56,7 +56,27 @@ func (c *Client) SetTimeout(timeout time.Duration) {
 	c.HTTPClient.Timeout = timeout
 }
 
+func extractContentTypeHeader(headers map[string]interface{}) (cType string){
+		headerType, ok := headers["Content-Type"]
+			if !ok {
+				return urlEncodedContentType
+			  } 
+			return headerType.(string)
```

Comment:
I believe this will panic if someone sets the "Content-Type" header to e.g. `[]string{"one", "two"}`.

It is perfectly legal HTTP to send two HTTP headers with the same key and even the same value.

Reply:
@kevinburkesegment How should this case be handled? Can you please add suggestions